include/dynd/strided_vals.hpp

```diff
@@ -8,284 +8,195 @@
 namespace dynd {
 
 struct start_stop_t {
-    intptr_t start;
-    intptr_t stop;
+  intptr_t start;
+  intptr_t stop;
 };
 
-namespace nd {
-
 namespace detail {
 
-template <int N>
-struct strided_utils {
-    static const char *get(const char *pointer, const intptr_t *index, const intptr_t *strides) {
-        return strided_utils<N - 1>::get(pointer, strides, index) + index[N - 1] * strides[N - 1];
-    }
-
-    static bool is_valid(const intptr_t *index, const start_stop_t *start_stop) {
-        return strided_utils<N - 1>::is_valid(index, start_stop) && (index[N - 1] >= start_stop[N - 1].start) && (index[N - 1] < start_stop[N - 1].stop);
-    }
+  template <typename T, bool inner>
+  class fixed_dim;
+
+  template <typename T>
+  class fixed_dim<T, true> {
+  public:
+    static const size_t arrmeta_size = sizeof(fixed_dim_type_arrmeta);
+    static const intptr_t ndim = 1;
+    typedef T dtype;
+
+    static int at(const intptr_t *i, const char *arrmeta, const char *data)
+    {
+      return *reinterpret_cast<const T *>(
+                 data +
+                 i[0] *
+                     reinterpret_cast<const fixed_dim_type_arrmeta *>(arrmeta)
+                         ->stride);
+    }
+
+    static void incr(const char *arrmeta, const char **data, intptr_t *index)
+    {
+      ++index[0];
+      *data +=
+          reinterpret_cast<const fixed_dim_type_arrmeta *>(arrmeta)->stride;
+    }
+  };
+
+  template <typename T>
+  class fixed_dim<T, false> {
```

Comment:
The flag is because the innermost details need to be specialized. But, yes, fixed_dim[var_dim[T]] and fixed_dim[fixed_dim[T]] should behave the same way at the outermost fixed_dim.